app/controllers/api/v1/sets_controller.rb

```diff
@@ -20,7 +28,25 @@ def create_uuids
         param_uuids.each { |uuid| Aker::Material.find_or_create_by!(id: uuid) }
       end
 
-      private
+      def clone
+        cloneparams = clone_params
+        set = Aker::Set.find(cloneparams[:set_id])
+        copy = set.clone(cloneparams[:name])
+        unless copy.save
+          return render json: { errors: [{ status: '422', title: 'Unprocessable entity', detail: 'The clone could not be created'}]}, status: :unprocessable_entity
+        end
+        # TODO: Ideally, render the complete JSONAPI version of the copy, with links, but I can't find any way to do that
+        render json: { data: copy }, status: :created
```

Comment:
You can use the ResourceSerializer from JSONAPI to build the a standard response. Think it'd be something like:

`JSONAPI::ResourceSerializer.new(Api::V1::SetResource).serialize_to_hash(Api::V1::SetResource.new(copy, nil))`

More info [in their docs](http://jsonapi-resources.com/v0.8/guide/serializer.html)